File: code/front end/admin-frontend/src/services/external-doctor-service.ts
import axiosClient from './http-client';

const BASE_URL = '/api/v1/external-doctors';

export const externalDoctorApi = {
  getSources: () => {
    return axiosClient.get(`${BASE_URL}/sources`);
  },

  getAllDoctors: () => {
    return axiosClient.get(BASE_URL);
  },

  createSource: (data: any) => {
    return axiosClient.post(`${BASE_URL}/sources`, data);
  },

  triggerSync: (sourceId: number) => {
    return axiosClient.post(`${BASE_URL}/sync/${sourceId}`);
  },

  approveDoctor: (id: number) => {
    return axiosClient.post(`${BASE_URL}/${id}/approve`);
  },

  deleteDoctor: (id: number) => {
    return axiosClient.delete(`${BASE_URL}/${id}`);
  },

  deleteAllDoctors: () => {
    return axiosClient.delete(BASE_URL);
  },

  deleteAll: () => {
    return axiosClient.delete('/external-doctors')
  },
  bulkApprove: (doctorIds: number[]) => {
    return axiosClient.post('/external-doctors/bulk-approve', doctorIds)
  },
  approveAll: () => {
    return axiosClient.post('/external-doctors/approve-all')
  }
};


export default externalDoctorApi;

File: code/backend/backend/src/main/java/com/caretriage/application/service/ExternalDoctorService.java
package com.caretriage.application.service;

import com.caretriage.domain.entity.external.ExternalDoctor;
import com.caretriage.domain.entity.external.ExternalDoctorImportJob;
import com.caretriage.domain.entity.external.ExternalDoctorSource;

import java.util.List;

public interface ExternalDoctorService {
    
    /**
     * Trigger a sync job from an external source API.
     */
    ExternalDoctorImportJob triggerSync(Long sourceId);

    /**
     * Get all external doctors.
     */
    List<ExternalDoctor> getAllExternalDoctors();
    
    /**
     * Find doctors by specialization.
     */
    List<ExternalDoctor> findBySpecialization(String specialization);

    /**
     * Approve an external doctor to be bookable.
     */
    ExternalDoctor approveDoctor(Long doctorId);

    List<ExternalDoctor> bulkApproveDoctors(List<Long> doctorIds);

    List<ExternalDoctor> approveAllUnverifiedDoctors();

    /**
     * Get all external sources
     */
    List<ExternalDoctorSource> getAllSources();

    /**
     * Create a new External Doctor Source
     */
    ExternalDoctorSource createSource(ExternalDoctorSource source);

    /**
     * Get preview of doctors imported by a specific job
     */
    List<ExternalDoctor> getJobPreview(Long jobId);

    /**
     * Find nearby sources based on location
     */
    List<ExternalDoctorSource> findNearbySources(Double lat, Double lng, Double radiusKm);

    /**
     * Delete an external doctor by ID
     */
    void deleteExternalDoctor(Long id);

    /**
     * Delete all external doctors
     */
    void deleteAllExternalDoctors();
}

